tools/rosconsole/src/rosconsole/rosconsole.cpp

```diff
@@ -394,8 +397,11 @@ void Formatter::print(void* logger_handle, ::ros::console::Level level, const ch
 
   fprintf(f, "%s\n", ss.str().c_str());
   
-  if (f == stdout) {
-    fflush(f);
+  if (g_force_stdout_line_buffered && f == stdout) {
```

Comment:
Should we have similar functionality for flushing `stderr`? Granted, `stdout` should be more common than `stderr`, but it seems odd to treat them differently.